### ISSUE TITLE
Unexpected nullable warning with default value for nullable enum parameter

### ISSUE THREAD

**Issue by MrTrillian** [dotnet/roslyn#52206]
**Version Used**:  3.9.0-3.21056.4 (e5eb71d6)

**Steps to Reproduce**:

`csc -nullable:enable Repro.cs`

```cs
void Foo(MyEnum? e = MyEnum.One) { }
Foo(); // CS8620
enum MyEnum { One = 1 }
```

**Expected Behavior**: No warnings.

**Actual Behavior**:
> Repro.cs(2,1): warning CS8620: Argument of type 'int' cannot be used for parameter 'e' of type 'MyEnum?' in 'void Foo(MyEnum? e = 1)' due to differences in the nullability of reference types.

That warning is invalid. There are no reference types involved, and the code is legal.

Does not repo when:
- Not using `-nullable:enable`
- Using an `int` parameter instead of an enum
- Using an enumerant with value `0` as the default argument value

**Comment by Youssef1313** [comment]
Duplicate of https://github.com/dotnet/roslyn/issues/51622 which was fixed in https://github.com/dotnet/roslyn/pull/51631.

**Comment by RikkiGibson** [comment]
Thank you for filing this issue. I will close based on @Youssef1313's observation above. Fix should be available in VS 16.10/.NET Core SDK 5.0.2xx.
